dbt_style_guide.md

```diff
@@ -1,237 +1,508 @@
+              or my_data.field_2 = 'ghi'
+          )
+      qualify row_number() over(
+          partition by my_data.field_1
+          order by my_data.start_date desc
+      ) = 1
+  )
 
-)
+  select * from final
+  ```
 
-select * from final
+## YAML and Markdown style guide
 
-```
+- Every subdirectory contains their own `.yml` file(s) which contain configurations for the models within the subdirectory.
 
-- Your join should list the "left" table first (i.e. the table you are selecting `from`):
-```sql
-select
-    trips.*,
-    drivers.rating as driver_rating,
-    riders.rating as rider_rating
+- YAML and markdown files should be prefixed with an underscore ( `_` ) to keep it at the top of the subdirectory.
 
-from trips
-left join users as drivers
-    on trips.driver_id = drivers.user_id
-left join users as riders
-    on trips.rider_id = riders.user_id
+- YAML and markdown files should be named with the convention `_<description>_<config>`.  
+  Examples: `_jaffle_shop_sources.yml`, `_jaffle_shop_docs.md`  
+  - `description` is typically the folder of models you're setting configurations for.  
+    Examples: `core`, `staging`, `intermediate`
+  - `config` is the top-level resource you are configuring.  
+    Examples: `docs`, `models`, `sources`
+- Indents should use two spaces.
 
-```
+- List items should be indented.
 
-## YAML style guide
+- Use a new line to separate list items that are dictionaries, where appropriate.
 
-* Indents should be two spaces
-* List items should be indented
-* Use a new line to separate list items that are dictionaries where appropriate
-* Lines of YAML should be no longer than 80 characters.
+- Lines of YAML should be no longer than 80 characters.
 
-### Example YAML
-```yaml
-version: 2
-
-models:
-  - name: events
-    columns:
-      - name: event_id
-        description: This is a unique identifier for the event
-        tests:
-          - unique
-          - not_null
-
-      - name: event_time
-        description: "When the event occurred in UTC (eg. 2018-01-01 12:00:00)"
-        tests:
-          - not_null
-
-      - name: user_id
-        description: The ID of the user who recorded the event
-        tests:
-          - not_null
-          - relationships:
-              to: ref('users')
-              field: id
-```
+- Items listed in a single .yml or .md file should be sorted alphabetically for ease of finding in larger files.
 
+- Each top-level configuration should use a separate `.yml` file (i.e, sources, models)
+  Example:
+  ```bash
+  models
+  ├── marts
+  └── staging
+      └── jaffle_shop
+          ├── _jaffle_shop_docs.md
+          ├── _jaffle_shop__models.yml
+          ├── _jaffle_shop__sources.yml
+          ├── stg_jaffle_shop__customers.sql
+          ├── stg_jaffle_shop__orders.sql
+          └── stg_jaffle_shop__payments.sql
+  ```
 
+### Example YAML
+  `_tpch_models.yml`:
+  ```yaml
+  version: 2
+
+  models:
+  
+    - name: base_tpch__nations

```

Comment:
```suggestion
    - name: base_jaffle_shop__nations

```

Reply:
Done - gonna refer you back to the commit history if you need to see since there's a lot of little ones here!